presentation/src/main/java/com/yapp/presentation/ui/member/qrcodescanner/QrCodeScanner.kt

```diff
@@ -82,12 +88,65 @@ fun QrCodeScanner(
                 .fillMaxWidth()
                 .fillMaxHeight()
         ) {
-            CameraPreview()
+            CameraPreview { code ->
+                try {
+                    val sessionId = AttendanceQrCodeParser.getSessionIdFromBarcode(code.rawValue)
+                    viewModel.setEvent(QrCodeContract.QrCodeUiEvent.SuccessToCheck(sessionId))
+                    Toast.makeText(context, sessionId.toString(), Toast.LENGTH_SHORT).show()
+                } catch (e: Exception) {
+                    Toast.makeText(context, "잘못된 코드입니다.", Toast.LENGTH_SHORT).show()
+                }
```

Comment:
1. setEvent 가 유저가 행동한 행위를 담아야 하는데, SuccessToCheck 는 유저가 행동한게 아니라 로직의 결과를 함축시킨 문장이어서 약간 와닿지 않는 듯 하네요! SuccessToCheck 로직이 추후에 어떤게 추가될 지는 모르겠지만 "유저가 QR 을 찍었다" 는 단어로 변경하던가, 아니면 체크 아이콘이 단발성으로 등장하니 setEffect 로 변경해볼 수 있을 것 같아요 
2. exception 안에도 setEffect 를 담아야 할듯요 ㅎㅎ 컴포저블 함수 내에서는 지금처럼 스스로 뷰를 띄우는 행위가 있어서는 안 되고, 어떠한 상태나 이벤트, 이펙트를 통해 뷰를 띄운다는 개념으로 가야할 듯 싶어요

Reply:
이름은 확실히 이상한 것 같네요! 적절한 이름으로 변경해 보겠습니다

요 부분이 아직까지 제가 확실하게 적응하지 못한 것 같아요 ㅠㅠ
지금 코드가
유저가 QR을 찍으면 -> try: 올바른 형식의 출석 코드인지 확인 -> 올바르다면 `setEvent()` 보냄 -> 뷰모델에서 해당 이벤트를 받으면 (현재는 로직이 없지만 로직이 적용되어서 디비에 결과를 반영한 뒤) `setState`로 `uiState`의 `attendanceState` 값을 `SUCCESS`로 변경 -> 뷰에서는 `SUCCESS`로 변경됨에 따라 체크 아이콘을 보여줌(when으로 STAND_BY, SUCCESS, COMPLETE 세 상태에 따라 다른 화면이 보여짐)
이렇게 되어 있는데 `setEffect`로 변경되어야 한다면 어떻게 구현해야 하는 것인가요? 🥲
`effect`를 `collect `하는 곳에서는 컴포저블 함수를 사용할 수 없도록 되어 있어서 어떻게 해야 뷰에 컴포저블 함수가 보여질 수 있도록 하는지? 잘 모르겠어요 ㅠㅠ
컴포저블 함수 내에 있는 `mutableState`의 값을 변경해서 그 값에 따라 뷰가 보이도록 변경하는 것이 맞을까요..?

Reply:
```suggestion
      viewModel.parseSeeionIdFromBarcode(sessionId)
```
그렇군요! 그럼 setState 로 하되, 뷰모델 안에 `parseSeeionIdFromBarcode` 함수를 두어서 지금 로직을 뷰모델로 옮기는 건 어떨까요?
뷰모델 안에서도 setEvent, setState, setEffect 를 설정할 수 있고, 컴포저블 함수에서는 이를 옵저빙하고 있으니 이벤트를 통해 뷰를 갱신시킬 수 있으니까요

Reply:
아하 답변 감사합니다!
현재 코드에서 setState가 뷰모델의 handleEvent()에 있으니 여기서 setState 하기 전에 파싱을 하고 디비 처리도 해 주면 될 것 같네요!
수정해 보겠습니다!! 구구절절 질문에 답변 정말 감사합니다 ㅠㅠ 🙏